skypy/pipeline/_pipeline.py

```diff
@@ -256,53 +209,42 @@ def write(self, file_format=None, overwrite=False):
                 filename = '.'.join((table, file_format))
                 self.state[table].write(filename, overwrite=overwrite)
 
-    def get_value(self, value):
-        '''return the value of a field
-
-        tuples specify function calls `(function name, function args)`
-        '''
+    def evaluate(self, value):
+        '''evaluate an item in the pipeline'''
 
-        if isinstance(value, dict):
-            # recurse dicts
-            return {k: self.get_value(v) for k, v in value.items()}
-        elif isinstance(value, list):
+        if isinstance(value, Sequence) and not isinstance(value, str):
             # recurse lists
-            return [self.get_value(v) for v in value]
-        elif isinstance(value, tuple):
-            # tuple (function, args, kwargs)
-            function, args, kwargs = value
-            return function(*self.get_value(args), **self.get_value(kwargs))
-        elif isinstance(value, str) and value[0] == '$':
-            # reference
-            return self[value[1:]]
+            return [self.evaluate(v) for v in value]
+        elif isinstance(value, Mapping):
+            # recurse dicts
+            return {k: self.evaluate(v) for k, v in value.items()}
+        elif isinstance(value, Item):
+            # evaluate item
+            return value.evaluate(self)
         else:
-            # plain value
+            # everything else return unchanged
             return value
 
-    def get_deps(self, args):
+    def depend(self, args):
         '''get dependencies from function args
 
         returns a list of all references found
         '''
 
-        if isinstance(args, str) and args[0] == '$':
-            # reference
-            return [args[1:]]
-        elif isinstance(args, tuple):
-            # recurse on function arguments
-            _, args, kwargs = args
-            return self.get_deps(args) + self.get_deps(kwargs)
-        elif isinstance(args, dict):
+        if isinstance(args, Sequence) and not isinstance(args, str):
+            # recurse list
+            return sum([self.depend(a) for a in args], [])
+        elif isinstance(args, Mapping):
             # get explicit dependencies
             deps = args.pop('.depends', [])
             # turn a single value into a list
-            if isinstance(deps, str) or not isinstance(deps, list):
+            if isinstance(deps, str) or not isinstance(deps, Sequence):
                 deps = [deps]
```

Comment:
```suggestion
            if isinstance(deps, str) or not isinstance(deps, Sequence):
                deps = Ref(deps).depend(self)
```
I think this does the same thing but wonder if it makes it clearer?

Reply:
I like that at the `Pipeline` level, all refs are `Ref` objects and not strings. But since this changes YAML syntax, let's do it in a separate changes.